agent/php_internal_instrument.c

```diff
@@ -734,6 +734,67 @@ NR_INNER_WRAPPER(mysqli_options) {
   }
 }
 
+/*
+ * Handle
+ *   bool mysqli_commit ( object $link [, int $flags=0, string $name] )
+ *   bool mysqli::commit ( [, int $flags=0, string $name] )
+ */
+NR_INNER_WRAPPER(mysqli_commit) {
+  zval* mysqli_obj = NULL;
+  nr_segment_t* segment = NULL;
+  nr_datastore_instance_t* instance = NULL;
+  nr_explain_plan_t* plan = NULL;
+  char* name = NULL;
+  char* sqlstr = NULL;
+  nr_string_len_t sqlstrlen;
+  zend_long flags = 0;
+  nr_string_len_t name_len = 0;
+  int zcaught = 0;
+
+  if (FAILURE
+      == zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
+                                  ZEND_NUM_ARGS() TSRMLS_CC, "o|ls",
+                                  &mysqli_obj, &flags, &name, &name_len)) {
+    if (FAILURE
+        == zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
+                                    ZEND_NUM_ARGS() TSRMLS_CC, "|ls", &flags,
+                                    &name, &name_len)) {
+      nr_wrapper->oldhandler(INTERNAL_FUNCTION_PARAM_PASSTHRU);
+      return;
+    } else {
+      mysqli_obj = NR_PHP_INTERNAL_FN_THIS();
+    }
+  }
+
+  if (NULL != name) {
+    sqlstr = nr_formatf("%s %s", "COMMIT", name);
+  } else {
+    sqlstr = nr_formatf("%s", "COMMIT");
+  }
+  sqlstrlen = nr_strlen(sqlstr);
+
+  segment = nr_segment_start(NRPRG(txn), NULL, NULL);
+
+  if (nrlikely(NULL != segment)) {
+    /*
+     * Set the stop time now so that we don't include the explain plan time.
+     */
+    segment->stop_time = nr_txn_now_rel(NRPRG(txn));
+    instance = nr_php_mysqli_retrieve_datastore_instance(mysqli_obj TSRMLS_CC);
+
+    nr_php_txn_end_segment_sql(&segment, sqlstr, sqlstrlen, plan,
+                               NR_DATASTORE_MYSQL, instance TSRMLS_CC);
+
+    nr_explain_plan_destroy(&plan);
+    nr_free(sqlstr);
```

Comment:
Should this free be outside of the if statement so we free `sqlstr` if `segment` is `NULL`?

Reply:
Good call.  I'll put it out of the `if` statement to ensure it is always freed.